### PR TITLE
API security system tests

### PR DESCRIPTION
**Pull request by christophe-papazian**
First version for API security system tests.

Support:
- "_dd.schema.req.headers": requests headers schema
- "_dd.schema.req.query": query parameters schema
- "_dd.schema.req.params": path parameters schema
- "_dd.schema.req.body": request body schema
- "_dd.schema.res.headers": response headers schema

Do not support yet:
- "_dd.schema.res.body": response body schema

Also : reformat `utils/_context/_scenarios.py` using black and isort

## Reviewer checklist

* [ ] If this PR modify anything else than sctriclty the default scenario, then remove the `run-default-scenario` label
* [ ] CI is green
   * [ ] If not, failing jobs are not related to this change (and you are 100% sure about this statement)

## Workflow

1. ⚠️⚠️ Create your PR as draft
2. Follows the style guidelines of this project (See [how to easily lint the code](https://github.com/DataDog/system-tests/blob/main/docs/edit/lint.md))
3. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
4. Mark it as ready for review

> **_NOTE:_**  By default in PR only default scenario tests will be launched. Temove `run-default-scenario` label to run all scenarios ([more info](https://datadoghq.atlassian.net/wiki/spaces/APMINT/pages/2866381467/CI+Workflow+Github+Actions))

Once your PR is reviewed, you can merge it ! :heart:

### CODE REVIEW

**Review by cbeauchesne — CHANGES_REQUESTED** [review]
linting is failing.

**Comment by christophe-papazian** [comment]
> linting is failing.

This is fixed

File: utils/_context/_scenarios.py [hunk 1]
```diff
@@ -844,6 +840,13 @@ class scenarios:
         },
     )
 
+    appsec_api_security = EndToEndScenario(
```
Comment (cbeauchesne) [inline utils/_context/_scenarios.py:843]: Could you add a comment explaining what this scenario do ?

**Review by cbeauchesne — CHANGES_REQUESTED** [review]
Could you add your scneario in the CI ?

File: utils/_context/_scenarios.py (hunk 1 above)
Comment (christophe-papazian) [reply utils/_context/_scenarios.py:843]: This is a scenario for the new feature/product **_API Security_**. I thought the name was explicit enough.

**Review by cbeauchesne — APPROVED** [review]

**Pull request merged**